app/controllers/mypages_controller.rb

```diff
@@ -0,0 +1,51 @@
+class MypagesController < ApplicationController
+
+  class Genre
+    attr_accessor :viewing, :movies, :percentage
+    def initialize(genre_id)
```

Comment:
`id` ではなく `ids` にすべきでしょう。

```suggestion
    def initialize(genre_ids)
```